yorkie/auth/webhook.go

```diff
@@ -49,29 +54,70 @@ func VerifyAccess(ctx context.Context, be *backend.Backend, info *types.AccessIn
 		return err
 	}
 
-	// TODO(hackerwins): We need to apply retryBackoff in case of failure
-	resp, err := http.Post(
-		be.Config.AuthorizationWebhookURL,
-		"application/json",
-		bytes.NewBuffer(reqBody),
-	)
-	if err != nil {
-		return err
-	}
-	defer func() {
-		if err := resp.Body.Close(); err != nil {
-			log.Logger.Error(err)
+	return withExponentialBackoff(be.Config, func() (int, error) {
+		resp, err := http.Post(
+			be.Config.AuthorizationWebhookURL,
+			"application/json",
+			bytes.NewBuffer(reqBody),
+		)
+		if err != nil {
+			return 0, err
 		}
-	}()
 
-	authResp, err := types.NewAuthWebhookResponse(resp.Body)
-	if err != nil {
+		defer func() {
+			if err := resp.Body.Close(); err != nil {
+				log.Logger.Error(err)
+			}
+		}()
+
+		if http.StatusOK != resp.StatusCode {
+			return resp.StatusCode, fmt.Errorf("%d: %w", resp.StatusCode, ErrUnexpectedStatusCode)
+		}
+
+		authResp, err := types.NewAuthWebhookResponse(resp.Body)
+		if err != nil {
+			return resp.StatusCode, err
+		}
+
+		if !authResp.Allowed {
+			return resp.StatusCode, fmt.Errorf("%s: %w", authResp.Reason, ErrNotAllowed)
+		}
+
+		return resp.StatusCode, nil
+	})
+}
+
+func withExponentialBackoff(config *backend.Config, webhookFn func() (int, error)) error {
+	statusCode, err := webhookFn()
+	if !shouldRetry(statusCode) {
 		return err
 	}
 
-	if !authResp.Allowed {
-		return fmt.Errorf("%s: %w", authResp.Reason, ErrNotAllowed)
+	var retries uint64
+	for {
+		if retries == config.AuthorizationWebhookMaxRetries {
+			return err
+		}
+
+		time.Sleep(waitInterval(retries, config.AuthorizationWebhookMaxWaitInterval))
+
+		statusCode, err := webhookFn()
+		if !shouldRetry(statusCode) {
+			return err
+		}
+		retries++
 	}
+}
+
+func waitInterval(retries uint64, maxWaitInterval uint64) time.Duration {
+	interval := math.Pow(2, float64(retries)) * 100
+	interval = math.Min(interval, float64(maxWaitInterval))
+	return time.Duration(int64(interval)) * time.Millisecond
+}
 
-	return nil
+func shouldRetry(statusCode int) bool {
+	return statusCode == http.StatusRequestTimeout ||
+		statusCode == http.StatusInternalServerError ||
+		statusCode == http.StatusBadGateway ||
+		statusCode == http.StatusServiceUnavailable
```

Comment:
Is there any reason to check these conditions? If there is no particular reason, why not start with the status below?

```suggestion
	return statusCode == http.StatusInternalServerError ||
		statusCode == http.StatusGatewayTimeout ||
		statusCode == http.StatusTooManyRequests
```

https://github.com/kubernetes/kubernetes/blob/fcf9300cc8e6918fce26e9ad1e9f7360a7cbf552/staging/src/k8s.io/apiserver/pkg/util/webhook/webhook.go#L63-L66

Reply:
@hackerwins Modifications to `auth_webhook_test.go` have been reflected. And I left a few more comments before proceeding with the rest of the modifications. Please check :)